packages/workbox-webpack-plugin/src/index.js

```diff
@@ -0,0 +1,108 @@
+const path = require('path');
+const getEntries = require('./lib/get-webpack-manifest-entries');
+const generateManifest = require('./lib/generate-webpack-manifest');
+const generateSW = require('./lib/generate-webpack-sw');
+const webpackAsset = require('./lib/utils/webpack-asset');
+const copyWorkboxSW = require('./lib/utils/copy-workbox-sw');
+const {setReadFile} = require('./lib/utils/read-file');
+
+/**
+ * Use the instance of this in the plugins array of the webpack config.
+ *
+ * @example
+ * const WorkboxBuildWebpackPlugin = require('workbox-webpack-plugin');
+ * .
+ * module.exports = {
+ *   ...
+ *   plugins: [
+ *    new WorkboxBuildWebpackPlugin({
+ *      chunks: ['main'],
+ *      excludeChunks: ['mutable'],
+ *      manifestFilename: string,
+ *      manifestVarName: '__file_manifest',
+ *      swSrc: './src/sw.js',
+ *     });
+ *   ]
+ * };
+ *
+ * @module workbox-webpack-plugin
+ */
+class WorkboxWebpackPlugin {
+  /**
+   * Creates an instance of WorkboxBuildWebpackPlugin.
+   *
+   * @param {Object} [config] All the options as passed to `workbox-build`.
```

Comment:
`{@link module:workbox-build~Configuration}` should, if I understand my JSDoc syntax, generate a link to the canonical description of all the `workbox-build` options.